### PR TITLE
Refactor ros init

### PR DESCRIPTION
**Pull request by kostyaplis**
Hi Duan, 

This PR address following issues:
1. Fix `dns.(sub)domain` references in `kubernetes.erb.tf`

2. Temporary commented  null-resource `k8s-tiller-rbac` dependancy on eks module. It caused me circular dependancy with `module.eks-cluster.module.eks.aws_launch_configuration.workers` when I launched `ros be infra apply` on existing cluster and it planned to update worker nodes config (volumes 50G -> 100G). 
I'm not sure if this dependancy is necessary, as  `k8s-tiller-rbac` triggered when kubeconfig file is ready...Does that implies that cluster is ready as well??
I'll keep it in mind and have a closer look when launching cluster infra from scratch

3. `ros be infra init` updated to consider AWS profile when getting and updating user kubeconfig
Precedence as following:
a. ENV AWS_PROFILE if set
b. `infra.config.cluster.aws_profile` if set
c. default AWS profile

### CODE REVIEW

File: lib/ros/be/infra/cluster/generator.rb [hunk 1]
```diff
@@ -30,7 +30,8 @@ def init_aws(cli)
                 STDOUT.puts "missing #{credentials_file}"
                 return
               end
-              cmd_string = "aws eks update-kubeconfig --name #{name}"
+              infra.config.cluster.aws_profile.nil? || ENV['AWS_PROFILE'] ? profile = "" : profile = "--profile #{infra.config.cluster.aws_profile}"
```
Comment (duanshiqiang) [inline lib/ros/be/infra/cluster/generator.rb]: `AWS_DEFAULT_PROFILE` is also very common to use, so we should check both.

File: lib/ros/be/infra/files/terraform/aws/eks-cluster/main.tf [hunk 2]
```diff
@@ -113,7 +114,7 @@ resource "aws_iam_role_policy_attachment" "eks-worker-external-dns" {
 }
 
 resource "null_resource" "k8s-tiller-rbac" {
-  depends_on = [module.eks]
+  #depends_on = [module.eks]
```
Comment (duanshiqiang) [inline lib/ros/be/infra/files/terraform/aws/eks-cluster/main.tf]: why we need to comment out this?

Comment (kostyaplis) [reply lib/ros/be/infra/files/terraform/aws/eks-cluster/main.tf]: Temporary commented null-resource k8s-tiller-rbac dependancy on eks module. It caused me circular dependancy with module.eks-cluster.module.eks.aws_launch_configuration.workers when I launched ros be infra apply on existing cluster and it planned to update worker nodes config (volumes 50G -> 100G).
I'm not sure if this dependancy is necessary, as k8s-tiller-rbac triggered when kubeconfig file is ready...Does that implies that cluster is ready as well??
I'll keep it in mind and have a closer look when launching cluster infra from scratch

File: lib/ros/be/infra/templates/terraform/aws/kubernetes.tf.erb [hunk 3]
```diff
@@ -62,7 +62,7 @@ locals {
   accelerator_name                    = "<%= infra.components.globalaccelerator&.config&.name.nil? ? infra.config.cluster.name : infra.components.globalaccelerator.config.name %>"
   iam_name                            = "<%= infra.components.iam&.config&.name.nil? ? infra.config.cluster.name : infra.components.iam.config.name %>"
   tags                                = var.tags
-  domain_name                         = "<%= infra.dns.sub_domain %><%= infra.dns.sub_domain.nil? ? "" : "."%><%= infra.dns.root_domain %>"
+  domain_name                         = "<%= infra.config.cluster.dns.subdomain %><%= infra.config.cluster.dns.subdomain.nil? ? "" : "."%><%= infra.config.cluster.dns.domain %>"
```
Comment (duanshiqiang) [inline lib/ros/be/infra/templates/terraform/aws/kubernetes.tf.erb]: I think you should not change this. the config files `production.yaml`, `production-uat.yaml` and `production-staging.yaml` set the DNS section at a different place.

We can remove our `eks-develop.yaml` config file.

Comment (kostyaplis) [reply lib/ros/be/infra/templates/terraform/aws/kubernetes.tf.erb]: You are right...confused between multiple configs...will push an update shortly

File: lib/ros/be/infra/cluster/generator.rb (hunk 1 above)
Comment (kostyaplis) [reply lib/ros/be/infra/cluster/generator.rb]: done

**Pull request merged**